packages/core/src/xr/XRManager.ts

```diff
@@ -0,0 +1,296 @@
+import { XRInputManager } from "./input/XRInputManager";
+import { XRSessionManager } from "./session/XRSessionManager";
+import { XRCameraManager } from "./feature/camera/XRCameraManager";
+import { XRSessionState } from "./session/XRSessionState";
+import { XRSessionMode } from "./session/XRSessionMode";
+import { XRFeature } from "./feature/XRFeature";
+import { IXRDevice } from "@galacean/engine-design/src/xr/IXRDevice";
+import { Engine } from "../Engine";
+import { Scene } from "../Scene";
+import { Entity } from "../Entity";
+
+type TConstructor<T extends new (engine: Engine, ...args: any[]) => XRFeature> = T extends new (
+  engine: Engine,
+  ...args: infer P
+) => XRFeature
+  ? P
+  : never;
+
+/**
+ * XRManager is the entry point of the XR system.
+ */
+export class XRManager {
+  /** Input manager for XR. */
+  inputManager: XRInputManager;
+  /** Session manager for XR. */
+  sessionManager: XRSessionManager;
+  /** Camera manager for XR. */
+  cameraManager: XRCameraManager;
+
+  /** @internal */
+  _xrDevice: IXRDevice;
+
+  private _engine: Engine;
+  private _scene: Scene;
+  private _origin: Entity;
+  private _mode: XRSessionMode = XRSessionMode.None;
+  private _features: XRFeature[] = [];
+
+  /**
+   * The current xr scene.
+   */
+  get scene(): Scene {
```

Comment:
Scene is an abstract concept. How does this design solve the problem of simultaneous rendering of multiple Scenes in the scene? I found that just using the interface in origin